### DISCUSSION TITLE
ROI detection workflow question.

### DISCUSSION THREAD

**Discussion by pr4deepr** [kushalkolar/MESmerize#70] [General]
Hi @kushalkolar 

I have a few questions about approaching the analysis in MESmerize. 

1. I have a question about using CNMF and detecting ROIs. When using CNMF and configuring the parameters, such as the `cnmf_kwargs `and `eval_kwargs`, is the idea that we test a few parameters and see what works for our dataset? 
Once we've found something that works, do we keep the parameters consistent across all our experimental replicates to make sure the ROI detection is consistent? 

2. The CNMF method doesn't always pick up cells that are bright and not active. If I'd like to quantify the proportion of cells that are active/inactive, how do I go about making sure CNMF detects these cells? Am I approaching this wrong?

Cheers
Pradeep

**Comment by kushalkolar** [comment]
Hi @pr4deepr , thanks for your questions!

1. After some experience I generally have an idea of a range of parameters that will work and then I make use of Mesmerize's script editor and batch manager to create a small gridsearch for each video. Depending on your recording setup and animal-to-animal variation the parameters might need to be optimized on a per-movie basis or per-animal basis, depends on what your data looks like.
2. Do you mean cells that are bright and don't really change their fluorescence throughout the imaging sessions? You could try seeding CNMF with a binary mask.

**Comment by pr4deepr** [comment]
Hi @kushalkolar 
   Thanks for the answers. 
1. Could you give a template for how you do the gridsearch? I'd be interested in how you set it up.  I've done a bit of scripting on MESMerize, but mainly to read in images and define metadata. 

2. Yes, thats a good idea. So, with the seeding, is it guaranteed that CNMF will pick an ROI? Assuming the binary should be 0 and 1, and not 0 and 255 like you get in FIJI/ImageJ?

Thanks heaps.

Cheers
Pradeep

**Reply by kushalkolar** [reply]
1. You can take a look at the examples provided on the doc pages for using the caiman motion correction and CNMF modules. They may need some modification.

2. I think it needs to be a np.bool, however this question is more about how CNMF itself works so I'd direct you to ask in the caiman gitter.

Let me know if you have further questions.
